### PR TITLE
[#4630] feat(iceberg): remove S3 SDK jar from Gravitino Iceberg REST server 

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
remove S3 SDK jar from Gravitino Iceberg REST server 

### Why are the changes needed?

Fix: #4630

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
tested on Iceberg REST server and Gravitino Iceberg catalog

### CODE REVIEW

**Comment by FANNG1** [comment]
@jerryshao please help to review this PR, and I want to cherry-pick it to branch-0.6

File: docs/iceberg-rest-service.md [hunk 1]
```diff
@@ -117,7 +117,7 @@ Gravitino Iceberg REST service supports using static access-key-id and secret-ac
 For other Iceberg s3 properties not managed by Gravitino like `s3.sse.type`, you could config it directly by `gravitino.iceberg-rest.s3.sse.type`.
 
 :::info
-Please set `gravitino.iceberg-rest.warehouse` to `s3://{bucket_name}/${prefix_name}` for Jdbc catalog backend, `s3a://{bucket_name}/${prefix_name}` for Hive catalog backend.
+Please set `gravitino.iceberg-rest.warehouse` to `s3://{bucket_name}/${prefix_name}` for Jdbc catalog backend, `s3a://{bucket_name}/${prefix_name}` for Hive catalog backend and download [Iceberg AWS bundle](https://mvnrepository.com/artifact/org.apache.iceberg/iceberg-aws-bundle) and place it to the classpath of Iceberg REST server.
```
Comment (jerryshao) [inline docs/iceberg-rest-service.md:120]: There're 3 things mixed in one sentence, which makes me so confused, 1) about jdbc backend, 2) about hive backend, 3) about aws bundle. You'd better separate into 3 sentence, and describe it more clearly.

File: docs/lakehouse-iceberg-catalog.md [hunk 2]
```diff
@@ -80,7 +80,7 @@ Supports using static access-key-id and secret-access-key to access S3 data.
 For other Iceberg s3 properties not managed by Gravitino like `s3.sse.type`, you could config it directly by `gravitino.bypass.s3.sse.type`.
 
 :::info
-Please set `gravitino.iceberg-rest.warehouse` to `s3://{bucket_name}/${prefix_name}` for JDBC catalog backend, `s3a://{bucket_name}/${prefix_name}` for Hive catalog backend.
+Please set `warehouse` to `s3://{bucket_name}/${prefix_name}` for JDBC catalog backend, `s3a://{bucket_name}/${prefix_name}` for Hive catalog backend and download [Iceberg AWS bundle](https://mvnrepository.com/artifact/org.apache.iceberg/iceberg-aws-bundle) and place it to `catalogs/lakehouse-iceberg/libs/`.
```
Comment (jerryshao) [inline docs/lakehouse-iceberg-catalog.md:83]: Also here.

File: docs/spark-connector/spark-catalog-iceberg.md [hunk 3]
```diff
@@ -97,44 +97,29 @@ DESC EXTENDED employee;
...
 
 :::info
 Iceberg catalog property `cache-enabled` is setting to `false` internally and not allowed to change.
 :::
 
+## Storage
+
+### S3
+
+To access the table stored on S3, you must add s3 secret in Spark configuration by `spark.sql.catalog.${iceberg_catalog_name}.s3.access-key-id` and `spark.sql.catalog.${iceberg_catalog_name}.s3.secret-access-key`, download [Iceberg AWS bundle](https://mvnrepository.com/artifact/org.apache.iceberg/iceberg-aws-bundle) and place it to the classpath of Spark.
```
Comment (yuqi1129) [inline docs/spark-connector/spark-catalog-iceberg.md:125]: the table -> tables.

> you must add s3 secret in Spark configuration by `spark.sql.catalog.${iceberg_catalog_name}.s3.access-key-id` and `spark.sql.catalog.${iceberg_catalog_name}.s3.secret-access-key`

You need to add s3 ... to the spark configuration using ... and .....,  additionally, download the ... and place it in spark's classpath.

File: docs/spark-connector/spark-catalog-iceberg.md [hunk 4]
```diff
@@ -97,44 +97,29 @@ DESC EXTENDED employee;
...
+| `jdbc-password`                 | `jdbc.password`                       | JDBC password                                                                                                                                                                                                       | 0.5.0         |
+| `io-impl`                       | `io-impl`                             | The io implementation for `FileIO` in Iceberg.                                                                                                                                                                      | 0.6.0         |
+| `s3-endpoint`                   | `s3.endpoint`                         | An alternative endpoint of the S3 service, This could be used for S3FileIO with any s3-compatible object storage service that has a different endpoint, or access a private S3 endpoint in a virtual private cloud. | 0.6.0         |
```
Comment (yuqi1129) [inline docs/spark-connector/spark-catalog-iceberg.md:112]: What's the meaning of `alternative`? Is this configuration optional?

Comment (FANNG1) [reply docs/spark-connector/spark-catalog-iceberg.md:112]: yes

Comment (FANNG1) [reply docs/spark-connector/spark-catalog-iceberg.md:112]: S3 had a global endpoint, and user doesn't need to set it by default

File: docs/iceberg-rest-service.md (hunk 1 above)
Comment (FANNG1) [reply docs/iceberg-rest-service.md]: updated

File: docs/lakehouse-iceberg-catalog.md (hunk 2 above)
Comment (FANNG1) [reply docs/lakehouse-iceberg-catalog.md]: updated

File: docs/iceberg-rest-service.md [hunk 5]
```diff
@@ -117,7 +117,7 @@ Gravitino Iceberg REST service supports using static access-key-id and secret-ac
 For other Iceberg s3 properties not managed by Gravitino like `s3.sse.type`, you could config it directly by `gravitino.iceberg-rest.s3.sse.type`.
 
 :::info
-Please set `gravitino.iceberg-rest.warehouse` to `s3://{bucket_name}/${prefix_name}` for Jdbc catalog backend, `s3a://{bucket_name}/${prefix_name}` for Hive catalog backend.
+To configure the JDBC catalog backend, set the `gravitino.iceberg-rest.warehouse` parameter to `s3://{bucket_name}/${prefix_name}`. For the Hive catalog backend, set `gravitino.iceberg-rest.warehouse` to `s3a://{bucket_name}/${prefix_name}`. Additionally, download the [Iceberg AWS bundle](https://mvnrepository.com/artifact/org.apache.iceberg/iceberg-aws-bundle) and place it in the classpath of Iceberg REST server.
```
Comment (jerryshao) [inline docs/iceberg-rest-service.md:120]: Does Iceberg aws bundle also include S3 related jars?

File: gradle/libs.versions.toml [hunk 6]
```diff
@@ -93,8 +92,6 @@ sun-activation-version = "1.2.0"
 error-prone = "3.1.0"
 
 [libraries]
-aws-s3 = { group = "software.amazon.awssdk", name = "s3", version.ref = "aws" }
-aws-sts = { group = "software.amazon.awssdk", name = "sts", version.ref = "aws" }
```
Comment (jerryshao) [inline gradle/libs.versions.toml:97]: Does Iceberg aws bundle also include these jars? Users don't need to add them to the classpath manually?

File: docs/iceberg-rest-service.md (hunk 5 above)
Comment (FANNG1) [reply docs/iceberg-rest-service.md:120]: yes, please refer https://github.com/apache/iceberg/blob/main/aws-bundle/build.gradle

File: gradle/libs.versions.toml (hunk 6 above)
Comment (FANNG1) [reply gradle/libs.versions.toml:97]: aws bundle includes these jars, and user need to add it manually.

**Comment by FANNG1** [comment]
@jerryshao @yuqi1129 , comments are addressed, please help to review again

**Review by jerryshao — APPROVED** [review]

**Pull request merged**